packages/react-app/src/contexts/BridgeContext.jsx

```diff
@@ -148,11 +154,35 @@ export const BridgeProvider = ({ children }) => {
     [setToken, fromToken, toToken],
   );
 
+  const checkForCustomToken = useCallback(async () => {
+    const { token } = isTokenInjected;
+    if (!fromToken && !toToken) {
+      const tokenList = await fetchTokenList(providerChainId);
```

Comment:
dont need to fetchTokenList, 
we can just call setToken with `token = {chainId, address}`

and handle any errors if the token is not found and throw a toast, and setDefaultToken after that.

Reply:
Im afraid I understood it right. Could you please rephrase it, as Im confused over fetching the contract address of the token without getting the list at the first place?

Reply:
I'm saying that the token param will be a token address so we can use it directly without needing to fetchTokenList

Reply:
Ahh, I see. So we don't have to get the list as we already have the most important info. Gotcha.